### PR TITLE
UIEH-1197: Jest+RTL > Test SettingsRootProxyRoute

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Add tests for `<SettingsRootProxyRoute>` component

## Issues
[UIEH-1197](https://issues.folio.org/browse/UIEH-1197)

### CODE REVIEW

File: src/routes/settings-root-proxy-route/settings-root-proxy-route.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+const propTypes = propTypes = {
+  confirmUpdateRootProxy: PropTypes.func.isRequired,
+  getProxyTypes: PropTypes.func.isRequired,
+  getRootProxy: PropTypes.func.isRequired,
+  history: ReactRouterPropTypes.history.isRequired,
+  match: ReactRouterPropTypes.match.isRequired,
+  proxyTypes: PropTypes.object.isRequired,
+  rootProxy: rootProxyShapes.RootProxyReduxStateShape.isRequired,
+  updateRootProxy: PropTypes.func.isRequired
```
Comment (mariiaschv) [inline src/routes/settings-root-proxy-route/settings-root-proxy-route.js:19]:
```suggestion
  updateRootProxy: PropTypes.func.isRequired,
```

File: src/routes/settings-root-proxy-route/settings-root-proxy-route.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  useEffect(() => {
+    getProxyTypes(match.params.kbId);
+    getRootProxy(match.params.kbId);
+  }, []);
+
+  useEffect(() => {
+    if (rootProxy.isUpdated) {
+      history.push({
+        pathname: `/settings/eholdings/${match.params.kbId}/root-proxy`,
+        state: { eholdings: true, isFreshlySaved: true }
```
Comment (mariiaschv) [inline src/routes/settings-root-proxy-route/settings-root-proxy-route.js:43]:
```suggestion
        state: { eholdings: true, isFreshlySaved: true },
```

File: src/routes/settings-root-proxy-route/settings-root-proxy-route.test.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    });
+
+    expect(mockGetProxyTypes).toHaveBeenCalledWith('test-kb-id');
+    expect(mockGetRootProxy).toHaveBeenCalledWith('test-kb-id');
+  });
+
+  describe('when kbId has changed', () => {
+    it('should request new data', async () => {
+      await act(async () => {
+        const { rerender } = await renderSettingsRootProxyRoute();
```
Comment (mariiaschv) [inline src/routes/settings-root-proxy-route/settings-root-proxy-route.test.js:104]: There should be an empty line after const

File: src/routes/settings-root-proxy-route/settings-root-proxy-route.test.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+  describe('when root proxy was updated', () => {
+    it('should call confirmUpdateRootProxy', async () => {
+      await act(async () => {
+        const { rerender } = await renderSettingsRootProxyRoute();
+        rerender(getSettingsRootProxyRoute({
+          rootProxy: {
+            isUpdated: true,
+          },
+        }));
```
Comment (mariiaschv) [inline src/routes/settings-root-proxy-route/settings-root-proxy-route.test.js:128]:
```suggestion
        const { rerender } = await renderSettingsRootProxyRoute();
        
        rerender(getSettingsRootProxyRoute({
          rootProxy: {
            isUpdated: true,
          },
        }));
```

**Review by mariiaschv — CHANGES_REQUESTED** [review]

**Review by Vladyslav-Velytskyi — APPROVED** [review]

File: src/routes/settings-root-proxy-route/settings-root-proxy-route.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { TitleManager } from '@folio/stripes/core';
+import { useIntl } from 'react-intl';
+
+import View from '../../components/settings/settings-root-proxy';
+
+import { rootProxy as rootProxyShapes } from '../../constants';
+
+const propTypes = propTypes = {
```
Comment (mariiaschv) [inline src/routes/settings-root-proxy-route/settings-root-proxy-route.js:11]:
```suggestion
const propTypes = {
```

**Review by mariiaschv — APPROVED** [review]

**Review by Vladyslav-Velytskyi — APPROVED** [review]

**Pull request merged**